pkg/cmd/gist/view/view.go

```diff
@@ -0,0 +1,109 @@
+package view
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/cli/cli/internal/ghinstance"
+	"github.com/cli/cli/pkg/cmd/gist/shared"
+	"github.com/cli/cli/pkg/cmdutil"
+	"github.com/cli/cli/pkg/iostreams"
+	"github.com/cli/cli/utils"
+	"github.com/spf13/cobra"
+)
+
+type ViewOptions struct {
+	IO         *iostreams.IOStreams
+	HttpClient func() (*http.Client, error)
+
+	Selector string
+	Raw      bool
+	Web      bool
+}
+
+func NewCmdView(f *cmdutil.Factory, runF func(*ViewOptions) error) *cobra.Command {
+	opts := &ViewOptions{
+		IO:         f.IOStreams,
+		HttpClient: f.HttpClient,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "view {<gist id> | <gist url>}",
+		Short: "View a gist",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.Selector = args[0]
+
+			if !opts.IO.IsStdoutTTY() {
+				opts.Raw = true
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return viewRun(opts)
+		},
+	}
+
+	cmd.Flags().BoolVarP(&opts.Raw, "raw", "r", false, "do not try and render markdown")
+	cmd.Flags().BoolVarP(&opts.Web, "web", "w", false, "open gist in browser")
+
+	return cmd
+}
+
+func viewRun(opts *ViewOptions) error {
+	gistID := opts.Selector
+
+	if opts.Web {
+		gistURL := gistID
+		if !strings.Contains(gistURL, "/") {
+			hostname := ghinstance.OverridableDefault()
+			gistURL = ghinstance.GistPrefix(hostname) + gistID
+		}
+		if opts.IO.IsStderrTTY() {
+			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(gistURL))
+		}
+		return utils.OpenInBrowser(gistURL)
+	}
+
+	u, err := url.Parse(opts.Selector)
+	if err == nil {
+		if strings.HasPrefix(u.Path, "/") {
+			gistID = u.Path[1:]
+		}
+	}
+
+	client, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+
+	gist, err := shared.GetGist(client, ghinstance.OverridableDefault(), gistID)
+	if err != nil {
+		return err
+	}
+
+	cs := opts.IO.ColorScheme()
+	if gist.Description != "" {
+		fmt.Fprintf(opts.IO.ErrOut, "%s\n", cs.Bold(gist.Description))
+	}
+
+	for filename, gistFile := range gist.Files {
+		fmt.Fprintf(opts.IO.ErrOut, "%s\n", cs.Gray(filename))
```

Comment:
I would imagine that I'd use `view` most of the time to display the raw contents of a single-file gist. In that case, printing the filename header would get in my way, and I'd be incentivized to `2>/dev/null`, but then I'd also miss out on any actual errors like those thrown by the API.

I'm imagining that a `-f, --filename` flag would be useful to display only the contents of a single file in a multi-file gist.

Lastly, printing alternately to stderr and stdout is not guaranteed to show up in a chronological fashion unless the streams are flushed between alternating.

All these factors make me feel that we should think more about the ideal way to print multi-file gists. 🤔

Reply:
- I can hide the filename for single file gists
- `-f` flag sounds great
- I'll just put everything on STDOUT; I was imagining some kind of "user wants to grep only the file output" scenario but it's not worth the race condition risk
- I couldn't think of a better way to view multi-file gist; I think this is a good place to start since I have a hypothesis that the vast majority of gists are single file (I scrolled through 10 years of my own gists and barely any had >1 file). We can explore an interactive "which file to view" picker like the "which file to edit" UI with a `--all` flag to just print everything.